### PR TITLE
fix some tests using reference mode stores

### PR DESCRIPTION
**Pull request by mmandlis**
some of #4843

### CODE REVIEW

File: src/runtime/manifest.ts [hunk 1]
```diff
@@ -1315,13 +1317,26 @@ ${e.message}
...
+        // Note: skip adding backing and container stores to context, if using reference-mode stores.
+        // TODO: there must be a better way!
+        return;
+      }
       this._addStore(store, tags);
     }
   }
 
+  isVolatileStore(store: UnifiedStore, tags: string[]): boolean {
+    if ((store.storageKey as StorageKey).protocol === VolatileStorageKey.protocol) return true;
```
Comment (shans) [inline src/runtime/manifest.ts]: you shouldn't have to cast storageKey to StorageKey here or on the next line.

File: src/runtime/manifest.ts [hunk 2]
```diff
@@ -1315,13 +1317,26 @@ ${e.message}
...
+    if ((store.storageKey as StorageKey).protocol === ReferenceModeStorageKey.protocol &&
+        (store.storageKey as ReferenceModeStorageKey).storageKey.protocol === VolatileStorageKey.protocol) {
```
Comment (shans) [inline src/runtime/manifest.ts]: might want to check both keys in the ReferenceModeStorageKey here.

File: src/runtime/manifest.ts [hunk 3]
```diff
@@ -1315,13 +1317,26 @@ ${e.message}
...
     // #volatile tag.
-    if (!this.findStoreById(store.id) &&
-        (store.storageKey as StorageKey).protocol !== 'volatile' &&
-        !tags.includes('volatile')) {
+    if (!this.findStoreById(store.id) && !this.isVolatileStore(store, tags)) {
+      if (Flags.defaultReferenceMode && !(store.storageKey instanceof ReferenceModeStorageKey)) {
+        // Note: skip adding backing and container stores to context, if using reference-mode stores.
+        // TODO: there must be a better way!
+        return;
+      }
```
Comment (shans) [inline src/runtime/manifest.ts]: why is this here? What's trying to add the backing & container stores? I know we add them to the arc to make sure serialization happens properly - is that being somehow copied into the manifest?

File: src/runtime/manifest.ts (hunk 2 above)
Comment (mmandlis) [reply src/runtime/manifest.ts]: done

File: src/runtime/manifest.ts (hunk 1 above)
Comment (mmandlis) [reply src/runtime/manifest.ts]: done
